kivy/ext/mpl/backend_kivy.py

```diff
@@ -0,0 +1,1013 @@
+                                              5, guiEvent=None)
+            else:
+                FigureCanvasBase.button_release_event(self, touch.x, touch.y,
+                                                    touch.button, guiEvent=None)
+            touch.ungrab(self)
+        else:
+            return super(FigureCanvasKivy, self).on_touch_up(touch)
+        return False
+
+    def keyboard_on_key_down(self, window, keycode, text, modifiers):
+        '''Kivy event to trigger matplotlib `key_press_event`.
+        '''
+        FigureCanvasBase.key_press_event(self, keycode[1], guiEvent=None)
+        return super(FigureCanvasKivy, self).keyboard_on_key_down(window,
+                                                    keycode, text, modifiers)
+
+    def keyboard_on_key_up(self, window, keycode):
+        '''Kivy event to trigger matplotlib `key_release_event`.
+        '''
+        FigureCanvasBase.key_release_event(self, keycode[1], guiEvent=None)
+        return super(FigureCanvasKivy, self).keyboard_on_key_up(window, keycode)
+
+    def _on_mouse_pos(self, *args):
+        '''Kivy Event to trigger the following matplotlib events:
+           `motion_notify_event`, `leave_notify_event` and
+           `enter_notify_event`.
+        '''
+        pos = args[1]
+        inside = self.collide_point(*pos)
+        FigureCanvasBase.motion_notify_event(self, pos[0], pos[1],
+                                                guiEvent=None)
+        if inside and self.inside_figure:
+            FigureCanvasBase.enter_notify_event(self, guiEvent=None, xy=None)
+            self.inside_figure = False
+        elif not inside and not self.inside_figure:
+            FigureCanvasBase.leave_notify_event(self, guiEvent=None)
+            self.inside_figure = True
+
+    def _on_size_changed(self, *args):
+        '''Changes the size of the matplotlib figure based on the size of the
+           widget. The widget will change size according to the parent Layout
+           size.
+        '''
+        w, h = self.size
+        dpival = self.figure.dpi
+        winch = w / dpival
+        hinch = h / dpival
+        self.figure.set_size_inches(winch, hinch)
+        self.draw()
+
+    def callback(self, *largs):
+        self.draw()
+
+    def close_event(self, guiEvent=None):
+        FigureCanvasBase.close_event(self, guiEvent=guiEvent)
+
+    def blit(self, bbox=None):
+        '''If bbox is None, blit the entire canvas to the widget. Otherwise
+           blit only the area defined by the bbox.
+        '''
+        self.blitbox = bbox
+
+    filetypes = FigureCanvasBase.filetypes.copy()
+    filetypes['png'] = 'Portable Network Graphics'
+
+    def print_png(self, filename, *args, **kwargs):
+        '''Call the widget function to make a png of the widget. It does a
+           one second delay to allow the widget to be completed.
+        '''
+        Clock.schedule_once(partial(self._print_image, filename), 1)
```

Comment:
That doesn't explain it. These are all blocking operation, how can there not be enough time?